File: models/entregaTipoLixo.js
module.exports = (sequelize, DataTypes) => {
  const EntregaTipoLixo = sequelize.define('EntregaTipoLixo', {
    kg: DataTypes.DECIMAL(8, 2),
  }, {
    tableName: 'tb_entrega_tipo_lixo',
    timestamps: false,
  });

  EntregaTipoLixo.associate = (models) => {
    EntregaTipoLixo.belongsTo(models.TipoLixo, { foreignKey: 'id_tipo__lixo' });
    EntregaTipoLixo.belongsTo(models.EntregaMaterial, { foreignKey: 'id_entrega_material' });
  };

  return EntregaTipoLixo;
};

File: src/main/java/org/acme/models/EntregaTipoLixo.java
package org.acme.models;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Entity
@Table(name = "tb_entrega_tipo_lixo")
public class EntregaTipoLixo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private BigDecimal kg;

    @ManyToOne
    @JoinColumn(name = "id_tipo__lixo")
    private TipoLixo tipoLixo;

    @ManyToOne
    @JoinColumn(name = "id_entrega_material")
    private EntregaMaterial entregaMaterial;
}
